frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -201,7 +205,6 @@ public void initChannel(Channel ch) {
                             (ChannelFutureListener)
                                     future -> {
                                         latch.countDown();
-                                        parentThreads.remove(WorkerThread.this); // NOPMD
                                         logger.info("{} Worker disconnected.", getWorkerId());
                                         retry();
```

Comment:
You can remote this retry() here, and move to end of run().
Instead, you can call shutdown(), but with a DISCONNECTED, or STOPPED state. shutdown will stop the thread and make while loop in run() break.